src/sync/chase_lev.rs

```diff
@@ -269,9 +283,11 @@ impl<T> Drop for Deque<T> {
     fn drop(&mut self) {
         let guard = epoch::pin();
 
-        let t = self.top.load(SeqCst);
-        let b = self.bottom.load(SeqCst);
-        let a = self.array.swap(None, SeqCst, &guard).unwrap();
+        // exclusive access here means that we only need to enforce sequential consistency once.
+        fence(SeqCst);
```

Comment:
A fence isn't required here. When drop is running you are already guaranteed to be the only thread still accessing the deque. This is guaranteed by fences in `Arc::drop`.